apollo-router/src/plugins/telemetry/config.rs

```diff
@@ -682,6 +683,37 @@ impl Conf {
             },
         )
     }
+
+    pub(crate) fn metrics_reference_mode(
+        configuration: &Configuration,
+    ) -> ApolloMetricsReferenceMode {
+        match configuration.apollo_plugins.plugins.get("telemetry") {
+            Some(telemetry_config) => {
+                match serde_json::from_value::<Conf>(telemetry_config.clone()) {
```

Comment:
I was talkin about this https://github.com/apollographql/router/blob/dev/apollo-router/src/router_factory.rs#L201:L201

Reply:
Is this still needed now that I'm only doing the initialisation once on startup and when the config is updated? I'm having trouble figuring out how to get a Box'd telemetry plugin without using the factory to create one like in that example. If it's still needed, could you let me know how I could use it or point me to some docs?

Reply:
Ah yes, I thought that in Configuration plugins were already instantiated but unfortunately I was wrong. As it's not longer in the hot path let's keep this code.